### PR TITLE
feat: implement rally onboard for local repos

### LINKED ISSUE #10: Implement rally onboard for local repos
## Description
- Implement `lib/onboard.js` (local path variant)
- Detect if cwd is a git repo
- Read config.yaml for team directory
- Create symlinks: `.squad/`, `.squad-templates/`, `.github/agents/squad.agent.md`
- Add entries to `.git/info/exclude`
- Register in projects.yaml
- Idempotent: skip existing symlinks
- UI: StatusMessage for each step

## Acceptance Criteria
- [ ] Creates all required symlinks
- [ ] Adds exclude entries
- [ ] Registers project in projects.yaml
- [ ] Idempotent on re-run

**Size:** M | **Owner:** Kaylee | **Depends on:** #2, #3, #4, #9

### PR DESCRIPTION
**Pull request by jsturtevant**
Closes #10

## Changes
- **lib/onboard.js**: Onboard command for local repos — detects git repo, reads config.yaml for team dir, creates symlinks (.squad/, .github/agents/squad.agent.md), adds .git/info/exclude entries, registers in projects.yaml. Idempotent on re-run.
- **bin/rally.js**: Wired `rally onboard [path]` as Commander subcommand (defaults to cwd).
- **test/onboard.test.js**: 11 tests covering all 4 acceptance criteria + 3 error cases.

## Acceptance Criteria
- [x] Creates all required symlinks
- [x] Adds exclude entries
- [x] Registers project in projects.yaml
- [x] Idempotent on re-run

## Test Results
All 81 tests pass (70 existing + 11 new).

### CODE REVIEW

File: test/onboard.test.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+  test('creates .github/agents/ parent directory if missing', async () => {
+    setupTeam();
+    const repoPath = createRepo(join(tempDir, 'my-repo'));
+
+    await onboard({ path: repoPath });
+
+    assert.ok(existsSync(join(repoPath, '.github', 'agents')), '.github/agents/ should exist');
+  });
+
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply test/onboard.test.js]: Fixed: Added `.squad-templates/` symlink creation to the symlinks array and added `.squad-templates/` to `setupTeam()` in tests. New test `creates .squad-templates symlink pointing to team dir` verifies the behavior.

File: lib/onboard.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    if (!existsSync(parentDir)) {
+      mkdirSync(parentDir, { recursive: true });
+    }
+
+    if (existsSync(linkPath)) {
+      console.log(`  ${linkRel} already exists — skipping`);
+    } else {
+      createSymlink(target, linkPath);
+      console.log(chalk.green('✓') + ` Symlinked ${linkRel} → ${target}`);
+    }
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply lib/onboard.js]: Fixed: When `linkPath` exists, we now validate it's a symlink pointing to the expected target. Three cases handled: (1) correct symlink → skip silently, (2) symlink to wrong target → warn with expected vs actual, (3) regular file/dir → warn it's not a symlink. Two new tests cover the warning cases.

**Review by jsturtevant — COMMENTED** [review]
## Mal — Code Review ✅

**Verdict: APPROVE** (posted as comment — formal approval blocked by single-user constraint)

### Acceptance Criteria
- [x] Creates all required symlinks (\.squad/, .squad-templates/, .github/agents/squad.agent.md)
- [x] Adds exclude entries to .git/info/exclude (including .worktrees/)
- [x] Registers project in projects.yaml
- [x] Idempotent on re-run — no duplicates, no throws

### Code Quality
- **Worktree detection**: `git rev-parse --git-dir` — correct approach, handles both regular repos and worktrees
- **Defensive parsing**: `readProjects() || { projects: [] }` and try/catch on `realpathSync` — both proper guards
- **Existing path validation**: Three-way check (correct symlink / wrong target / not a symlink) with clear warnings
- **Windows compat**: `statSync(target).isFile()` to choose file vs junction type — smart auto-detect
- **Test coverage**: 16 tests covering all 4 acceptance criteria + 5 edge cases from Copilot review

### Copilot Review Comments
All 7 comments addressed by Wash in commit 028084e. Each reply references the specific fix.

### CI
All 86 tests passing on Node 18/20/22.

**No blocking issues. Ready to merge.**
